agent/trufflehog_agent.py

```diff
@@ -50,40 +49,48 @@ def _process_scanner_output(self, output: bytes) -> list[dict[str, Any]]:
         secrets = utils.prune_reports(secrets)
         return secrets
 
-    def _run_scanner(self, input_file: str) -> bytes | None:
+    @staticmethod
+    def run_scanner(input_type: str, input_media: str) -> bytes | None:
         try:
             cmd_output = subprocess.check_output(
-                ["trufflehog", "filesystem", input_file, "--only-verified", "--json"]
+                ["trufflehog", input_type, input_media, "--only-verified", "--json"]
             )
-        except subprocess.CalledProcessError:
+        except subprocess.CalledProcessError as e:
+            logger.error("Error : %s", e)
             return None
         return cmd_output
 
     def process(self, message: m.Message) -> None:
-        """Runs the trufflehog tool ont the file/link recieved.
+        """
+        Runs the trufflehog tool ont the file/link recieved.
 
         Args:
             message: the message containing the trufflehog tool input file.
 
         Returns:
             None.
         """
-        logger.info("Processing input.")
-
-        with tempfile.NamedTemporaryFile() as target_file:
-            target_file.write(message.data.get("content", b""))
-            target_file.seek(0)
-            input_media = target_file.name
-
-            logger.info("Starting trufflehog tool.")
-
-            cmd_output = self._run_scanner(input_media)
-            if cmd_output is None:
-                return
-
-            logger.info("Parsing trufflehog output.")
-
-            secrets = self._process_scanner_output(cmd_output)
+        logger.info("Processing input and Starting trufflehog.")
+
+        if message.selector == "v3.asset.link":
+            link = message.data.get("url", "")
+            link_type = process_input.process_link(link)
+            cmd_output = self.run_scanner(str(link_type), link)
+        elif message.selector == "v3.asset.file":
+            cmd_output = process_input.process_file(message.data.get("content", b""))
+        elif message.selector == "v3.capture.logs":
+            cmd_output = process_input.process_file(message.data.get("message", b""))
+        elif message.selector == "v3.capture.request_response":
```

Comment:
Both of these don't have unit tests.

Reply:
both of them are handled as files, it would have the exact same behavior and steps.

Reply:
yes I see the issue, [2d6bca5](https://github.com/Ostorlab/agent_trufflehog/pull/2/commits/2d6bca55ffcf651236e81ab0e186046504a2205a) added tests.

Reply:
I got confused, [923d206](https://github.com/Ostorlab/agent_trufflehog/pull/2/commits/923d2064eebc1cc720787183987656f4e951e054) added.